common/flatpak-utils.c

```diff
@@ -2124,9 +2126,19 @@ flatpak_summary_match_subrefs (GVariant          *summary_v,
           id_start = strchr (cur, '/');
           if (id_start == NULL)
             continue;
+          id_start += 1;
+
+          id_end = strchr (id_start, '/');
+          if (id_end == NULL)
+            continue;
 
           /* But only prefix of id */
-          if (!g_str_has_prefix (id_start + 1, parts_prefix))
+          if (!g_str_has_prefix (id_start, parts_prefix))
+            continue;
+
+          /* And no dots (ie only one level of subelements */
```

Comment:
I think the comment would have been better if it mentioned why: We want to install prefix.$ID, but not prefix.$ID.Sources